### ISSUE TITLE
Autotools/buildsys build fails by default

### ISSUE THREAD

**Issue by angband-import** [angband/angband#3692]
**Reported by myshkin on 4 Mar 2012 22:25 UTC**
The default settings to configure (i.e. sdl-mixer enabled and sdl disabled) fail to add -lSDL to the LIBS, but snd-sdl.c relies on functions in that library (specifically, SDL_RWFromFile(), via the Mix_LoadWAV macro), at least in the version of SDL I have. Reported by bacchist in #angband-dev.

**Comment by angband-import** [comment]
**Attachment from myshkin on 19 Aug 2012 17:47 UTC**
Transcript of build session
REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1625_GIT_3692/typescript

**Comment by angband-import** [comment]
**Attachment from myshkin on 19 Aug 2012 17:59 UTC**

REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1625_GIT_3692/config.log

**Comment by angband-import** [comment]
**Comment by myshkin on 4 Mar 2012 22:30 UTC**
Possibly related to #1153.

**Comment by angband-import** [comment]
**Comment by magnate on 5 Mar 2012 18:14 UTC**
Surely the solution here is to change the default sdl-mixer config to disabled?

(Changing milestone to 3.4 as per other buildsys tickets - not specific to v4.)

**Comment by angband-import** [comment]
**Comment by myshkin on 5 Mar 2012 18:59 UTC**
It is possible that we should do so, but that won't fix the problem, unless we also add a dependency such that `--enable-sdl-mixer` requires `--enable-sdl`.

**Comment by angband-import** [comment]
**Modified by magnate on 6 Apr 2012 14:50 UTC**

**Comment by angband-import** [comment]
**Comment by pnd10 on 19 Aug 2012 11:44 UTC**
Hmmm. Works for me on debian (wheezy). I did:
```
$ ./autogen.sh
$ ./configure --with-no-install
$ make clean; make
```
and it built fine.

myshkin - can you post a trace of what fails for you?

**Comment by angband-import** [comment]
**Comment by myshkin on 19 Aug 2012 17:47 UTC**
noz, see attached, from commit 173d165 of master.

**Comment by angband-import** [comment]
**Comment by myshkin on 19 Aug 2012 18:01 UTC**
I should note that I'm running Ubuntu 11.10 (Oneiric Ocelot).

The link commands and output:
```
gcc -o angband angband.o main.o main-gcu.o main-x11.o snd-sdl.o   -L/usr/lib -lncursesw -ltinfo  -lSM -lICE  -lX11  -lSDL_mixer -lm
/usr/bin/ld: snd-sdl.o: undefined reference to symbol 'SDL_GetError'
/usr/bin/ld: note: 'SDL_GetError' is defined in DSO /usr/lib/libSDL-1.2.so.0 so try adding it to the linker command line
/usr/lib/libSDL-1.2.so.0: could not read symbols: Invalid operation
collect2: ld returned 1 exit status
```

**Comment by angband-import** [comment]
**Comment by myshkin on 19 Aug 2012 22:41 UTC**
Fixed in commit 955f77c in master.
